<launch>
    <!-- Start the hokuyo laser at the base -->
    <node name="base_laser" pkg="hokuyo_node" type="hokuyo_node" respawn="false" output="screen">
        <remap from="scan" to="base_scan/scan"/>
        <param name="port" value="/dev/wubble2BaseLaser"/>
        <param name="min_ang" value="-1.65"/>
        <param name="max_ang" value="1.65"/>
        <param name="frame_id" value="base_scan_link"/>
    </node>

    <!-- Filter for base laser shadowing/veiling -->
    <node pkg="laser_filters" type="scan_to_cloud_filter_chain" respawn="true" name="base_shadow_filter" >
        <remap from="scan" to="base_scan/scan" />
        <remap from="cloud_filtered" to="base_scan/shadow_filtered" />
        <param name="target_frame" value="base_footprint" />
        <param name="high_fidelity" value="false" />
        <rosparam command="load" file="$(find wubble_description)/params/shadow_filter.yaml" />
    </node>

    <!-- Filter for base laser scans that hit the body of the robot -->
    <node pkg="robot_self_filter" type="self_filter" name="base_laser_self_filter" respawn="true" output="screen" >
        <remap from="cloud_in" to="base_scan/shadow_filtered" />
        <remap from="cloud_out" to="base_scan/marking" />
        <param name="sensor_frame" type="string" value="base_scan_link" />
        <rosparam file="$(find wubble_description)/params/w2_base_self_filter.yaml" command="load" />
    </node>
</launch>
